core/src/main/java/org/infinispan/protostream/impl/TagWriterImpl.java

```diff
@@ -794,6 +819,51 @@ void writeBytes(ByteBuffer value) throws IOException {
          }
       }
 
+      @Override
+      void writeUTF8Field(int number, String s) throws IOException {
+         if (!(out instanceof LazyByteArrayOutputStream os)) {
+            super.writeUTF8Field(number, s);
+            return;
+         }
+
+         int strlen = s.length();
+         int tag = WireType.makeTag(number, WireType.WIRETYPE_LENGTH_DELIMITED);
+         int varIntTagLen = varIntSize(tag);
+         int varIntBytesLen = varIntSize(strlen);
+         int startPos = os.getPosition();
+
+         // First optimize for 1 - 127 case
+         os.ensureCapacity(startPos + varIntTagLen + varIntBytesLen + strlen);
```

Comment:
why not assume the worst case of 5 bytes per varInt to avoid computing it? With the array growing double in size, I don't expect it to create growth more often.

Reply:
On my local benchmarks calculating the size resulted in better performance than pre-allocating.